ultratrace2/model/files/bundle.py

```diff
@@ -41,30 +41,43 @@ class FileBundleList:
             ".git",
             "node_modules",
             "__pycache__",
+            ".ultratrace",
             # FIXME: add more ignoreable dirs
         ]
     )
 
-    def __init__(self, path: str, extra_exclude_dirs: List[str] = []):
+    def __init__(self, bundles: Dict[str, FileBundle]):
 
-        # FIXME: implement `extra_exclude_dirs` as a command-line arg
-        for extra_exclude_dir in extra_exclude_dirs:
-            self.exclude_dirs.add(extra_exclude_dir)
+        self.current_bundle = None
+        self.bundles: Dict[str, FileBundle] = bundles
 
-        self.path = path
-        self.has_alignment_impl = False
-        self.has_image_impl = False
-        self.has_sound_impl = False
+        self.has_alignment_impl: bool = False
+        self.has_image_impl: bool = False
+        self.has_sound_impl: bool = False
 
-        self.current_bundle = None
+        for filename, bundle in bundles.items():
+            if not self.has_alignment_impl and bundle.alignment_file.has_impl():
+                self.has_alignment_impl = True
+            if not self.has_image_impl and bundle.image_file.has_impl():
+                self.has_image_impl = True
+            if not self.has_sound_impl and bundle.sound_file.has_impl():
+                self.has_sound_impl = True
+
+    @classmethod
+    def build_from_dir(
+        cls, root_path: str, extra_exclude_dirs: Sequence[str] = []
+    ) -> "FileBundleList":
+
+        # FIXME: implement `extra_exclude_dirs` as a command-line arg
+        exclude_dirs = cls.exclude_dirs.copy().union(extra_exclude_dirs)
```

Comment:
`.union()` returns a new set (c.f. https://docs.python.org/3.7/library/stdtypes.html#frozenset.union)
```suggestion
        exclude_dirs = cls.exclude_dirs.union(extra_exclude_dirs)
```

Reply:
The doc you linked is for `FrozenSet`

Reply:
Although it probably would be better to use `FrozenSet` anyway ...

Reply:
Done f0f6154